scripts/generate-unicode-id-parts.js

```diff
@@ -0,0 +1,158 @@
+    }
+    return text;
+  }
+}
+
+class Writer {
+  constructor() {
+    /** @type {number} */
+    this.lastCodePoint = -1;
+    /** @type {NonSurrogateRange[]} */
+    this.nonSurrogateRanges = [];
+    /** @type {LeadSurrogateRange[]} */
+    this.surrogateRanges = [];
+    /** @type {NonSurrogateRange} */
+    this.nonSurrogateRange;
+    /** @type {LeadSurrogateRange} */
+    this.leadSurrogateRange;
+    /** @type {TrailSurrogateRange} */
+    this.trailSurrogateRange;
+  }
+
+  /** @param {number} codePoint */
+  push(codePoint) {
+    if (codePoint <= this.lastCodePoint)
+      throw new Error('Code points must be added in order.');
+    this.lastCodePoint = codePoint;
+
+    if (codePoint < MAX_UNICODE_NON_SURROGATE) {
+      if (
+        this.nonSurrogateRange &&
+        this.nonSurrogateRange.lastCodePoint === codePoint - 1
+      ) {
+        this.nonSurrogateRange.lastCodePoint = codePoint;
+        return;
+      }
+      this.nonSurrogateRange = new NonSurrogateRange(codePoint);
+      this.nonSurrogateRanges.push(this.nonSurrogateRange);
+    } else {
+      const leadSurrogate = Math.floor((codePoint - 0x10000) / 0x400) + 0xd800;
+      const trailSurrogate = ((codePoint - 0x10000) % 0x400) + 0xdc00;
+      if (
+        !this.leadSurrogateRange ||
+        this.leadSurrogateRange.leadSurrogate !== leadSurrogate
+      ) {
+        this.trailSurrogateRange = undefined;
+        this.leadSurrogateRange = new LeadSurrogateRange(leadSurrogate);
+        this.surrogateRanges.push(this.leadSurrogateRange);
+      }
+
+      if (
+        this.trailSurrogateRange &&
+        this.trailSurrogateRange.lastTrailSurrogate === trailSurrogate - 1
+      ) {
+        this.trailSurrogateRange.lastTrailSurrogate = trailSurrogate;
+        return;
+      }
+
+      this.trailSurrogateRange = new TrailSurrogateRange(trailSurrogate);
+      this.leadSurrogateRange.ranges.push(this.trailSurrogateRange);
+    }
+  }
+
+  toString() {
+    let first = this.nonSurrogateRanges.join('');
+    let second = this.surrogateRanges.join('|');
+    return first && second
+      ? `([${first}]|${second})`
+      : first
+      ? `[${first}]`
+      : second
+      ? `(${second})`
+      : '';
+  }
+}
+
+const MAX_UNICODE_NON_SURROGATE = 0xffff;
+const MAX_UNICODE_CODEPOINT = 0x10ffff;
+const isStart = c => /\p{ID_Start}/u.test(c);
+const isContinue = c => /\p{ID_Continue}/u.test(c);
+
+let idStartWriter = new Writer();
+let idContinueWriter = new Writer();
+
+for (let cp = 0; cp < MAX_UNICODE_CODEPOINT; cp++) {
+  const ch = String.fromCodePoint(cp);
+  if (isStart(ch)) {
+    idStartWriter.push(cp);
+  }
+  if (isContinue(ch)) {
+    idContinueWriter.push(cp);
+  }
+}
+
+console.log(`/**
+* Generated by scripts/generate-unicode-id-parts.js on node ${
+  process.version
+} with unicode ${process.versions.unicode}
+* based on http://www.unicode.org/reports/tr31/ and https://www.ecma-international.org/ecma-262/6.0/#sec-names-and-keywords
```

Comment:
Consider pointing to the latest spec draft at https://tc39.es/ecma262/ instead of a versioned snapshot